docs/core/tutorials/cli-templates-create-item-template.md

```diff
@@ -0,0 +1,227 @@
+
+* Open a terminal and navigate to the _working\templates\\_ folder.
+
+## Create an item template
+
+An item template is a specific type of template that produces one or more files. These types of templates are useful when you want to generate something like a config, code, or solution file. In this example, you'll create a class that adds an extension method to the string type.
+
+Open a terminal and navigate to the _working\templates\\_ folder and create a new subfolder named _extensions_. Enter the folder.
+
+```console
+working
+└───templates
+    └───extensions
+```
+
+Create a new file named _CommonExtensions.cs_ and open it with your favorite text editor. This class will provide an extension method named `Reverse` that reverses the contents of a string. Paste in the following code and save the file:
+
+```csharp
+using System;
+
+namespace System
+{
+    public static class StringExtensions
+    {
+        public static string Reverse(this string value)
+        {
+            var tempArray = value.ToCharArray();
+            Array.Reverse(tempArray);
+            return new string(tempArray);
+        }
+    }
+}
+```
+
+Now that you have the content of the template created, you need to create the template config at the root folder of the template.
+
+## Create the template config
+
+Templates are recognized in .NET Core by a special folder and config file that exist at the root of your template. In this tutorial, your template folder is located at _working\templates\extensions\\_.
+
+When you create a template, all files and folders in the template folder are included as part of the template except for the special config folder. This config folder is named _.template.config_.
+
+First, create a new subfolder named _.template.config_, enter it. Then, create a new file named _template.json_. Your folder structure should look like this:
+
+```console
+working
+└───templates
+    └───extensions
+        └───.template.config
+                template.json
+```
+
+Open the _template.json_ with your favorite text editor and paste in the following json code and save it:
+
+```json
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Me",
+  "classifications": [ "Common", "Code" ],
+  "identity": "ExampleTemplate.StringExtensions",
+  "name": "Example templates: string extensions",
+  "shortName": "stringext",
+  "tags": {
+    "language": "C#",
+    "type": "item"
+  }
+}
+```
+
+This config file contains all of the settings for your template. You can see the basic settings such as `name` and `shortName` but also there's a `tags/type` value that is set to `item`. This categorizes your template as an item template. There's no restriction on the type of template you create. The `item` and `project` values are common names that .NET Core recommends so that users can easily filter the type of template they're searching for.
```

Comment:
```suggestion
This config file contains all the settings for your template. You can see the basic settings, such as `name` and `shortName`, but there's also a `tags/type` value that is set to `item`. This categorizes your template as an item template. There's no restriction on the type of template you create. The `item` and `project` values are common names that .NET Core recommends so that users can easily filter the type of template they're searching for.
```